p2p/src/sync/peer_context.rs

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) 2022 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::BTreeSet;
+
+use common::{
+    chain::block::{Block, BlockHeader},
+    primitives::Id,
+};
+
+// TODO: Investigate if we need some kind of "timeouts" (waiting for blocks or headers).
+pub struct PeerContext {
```

Comment:
It's probably useful to track the current best block for the peer and the current unknown best block.